File: Leetcode/Easy/38-N-th-Tribonacci-Number/solution.js
var tribonacci = function(n) {
    let cache  = {};
    let tri = (x) => {
        if (cache[x]) return cache[x];
        if (x < 2) return x;
        if (x == 2) return 1;
        return cache[x] = tri(x - 1) + tri(x - 2) + tri(x - 3)
    }
    return tri(n);
}

File: Leetcode/Easy/38-N-th-Tribonacci-Number/solution.py
class Solution:
    def tribonacci(self, n: int) -> int:
        cache = {}
        def fast(x):
            if cache.get(x):return cache.get(x)
            if x < 2:return x
            if x == 2:return 1
            cache[x] = fast(x-3) + fast(x-2) + fast(x-1)
            return cache[x]
        return fast(n)
            